rustls/src/vecbuf.rs

```diff
@@ -154,6 +151,23 @@ impl ChunkVecBuffer {
         self.consume(used);
         Ok(used)
     }
+
+    /// Fallback implementation of write_to that does not depend on Vectored I/O
+    /// however, it consumes all the available chunks
+    #[cfg(not(feature = "std"))]
+    pub(crate) fn write_to(&mut self, wr: &mut dyn io::Write) -> io::Result<usize> {
+        if self.is_empty() {
+            return Ok(0);
+        }
+
+        let mut used = 0;
+        for chunk in self.chunks.iter() {
+            used += wr.write(&chunk)?;
```

Comment:
This looks suspicious. What if the number of bytes written is less than the number returned?

Reply:
Hmm, maybe we should use write_all for that.

Reply:
Note that the original write_vectored definition can also have this hazard (https://doc.rust-lang.org/std/io/trait.Write.html#method.write_vectored) except it only affects the first 64 chunks of taken buffer.

Reply:
What I meant is what happens in this case:

1. Chunk 1: 16 bytes, Written: 16
2. Chunk 2: 16 bytes, Written 8
3. Chunk 3: 16 bytes, Written 8

You lost the last 8 bytes of chunk 2 and you'll write chunk 3 again on the next `write_to` call.

The third write in this example should never happen.

Reply:
Yep, I was think of this kind of situation exactly. So I think adding the chunk length after `write_all`, then that should be safe.